pkg/instancetype/instancetype.go

```diff
@@ -800,6 +815,50 @@ func applyCPU(field *k8sfield.Path, instancetypeSpec *instancetypev1beta1.Virtua
 	return nil
 }
 
+func validateCPU(field *k8sfield.Path, instancetypeSpec *instancetypev1beta1.VirtualMachineInstancetypeSpec, vmiSpec *virtv1.VirtualMachineInstanceSpec) (conflicts Conflicts) {
+	if _, hasCPURequests := vmiSpec.Domain.Resources.Requests[k8sv1.ResourceCPU]; hasCPURequests {
+		conflicts = append(conflicts, field.Child("domain", "resources", "requests", string(k8sv1.ResourceCPU)))
+	}
+
+	if _, hasCPULimits := vmiSpec.Domain.Resources.Limits[k8sv1.ResourceCPU]; hasCPULimits {
+		conflicts = append(conflicts, field.Child("domain", "resources", "limits", string(k8sv1.ResourceCPU)))
+	}
+
+	if vmiSpec.Domain.CPU.Sockets != 0 {
+		conflicts = append(conflicts, field.Child("domain", "cpu", "sockets"))
+	}
+
+	if vmiSpec.Domain.CPU.Cores != 0 {
+		conflicts = append(conflicts, field.Child("domain", "cpu", "cores"))
+	}
+
+	if vmiSpec.Domain.CPU.Threads != 0 {
+		conflicts = append(conflicts, field.Child("domain", "cpu", "threads"))
+	}
+
+	if vmiSpec.Domain.CPU.Model != "" && instancetypeSpec.CPU.Model != nil {
+		conflicts = append(conflicts, field.Child("domain", "cpu", "model"))
+	}
+
+	if vmiSpec.Domain.CPU.DedicatedCPUPlacement && instancetypeSpec.CPU.DedicatedCPUPlacement != nil {
```

Comment:
Should we fail also if vmiSpec and instancetypeSpec match?

Reply:
I think the fact that it's set in both the VMI and instance type is enough here?

We can't check the negative case for `vmiSpec.Domain.CPU.DedicatedCPUPlacement` anyway because it's using a `bool` that defaults to `false` so we just assume we can overwrite it here.

Reply:
Got you! Just for information: By design, istancetype and vmi specifications cannot conflict in any field?

Reply:
The specific fields related to each other between the instance type and VMI yes that's correct. I'll write up the negative `vmiSpec.Domain.CPU.DedicatedCPUPlacement` case as a bug as we could easily switch that (and likely other attributes) over to a pointer bool to improve our handling of this.

Reply:
Thank you! :+1: